core/drivers/nvmem/nvmem_huk.c

```diff
@@ -29,6 +29,8 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 
 	memcpy(hwkey->data, huk, HW_UNIQUE_KEY_LENGTH);
 
+	return res;
```

Comment:
I see an issue here. Each call to `tee_otp_get_hw_unique_key()` will allocate a new instance of the NVMEM cell data. It would be much more simple to fully prepare a reliable `huk_cell` reference from `nvmem_huk_probe()`.